src/datasets.py

```diff
@@ -1,15 +1,31 @@
 import logging
-from typing import Dict, List, Tuple
+from typing import TypeAlias
 
 import numpy as np
 import torch
 import xarray as xr
 from einops import rearrange
+from jaxtyping import Float, Integer
 from torch.utils.data import Dataset
 
+from constants import (
+    Extra,
+    ExtraVars,
+    GridMask,
+    Input,
+    InputMask,
+    InputVars,
+    Label,
+    OutputVars,
+    TotalInput,
+)
 from utils.data import Normalize
 from utils.device import get_device, using_gpu
 
+TDInput: TypeAlias = TotalInput | Float[TotalInput, "*batch"]
+TDLabel: TypeAlias = Label | Float[Label, "*batch"]
```

Comment:
```suggestion
TDInput: TypeAlias = Float[TotalInput, "*batch"]
TDLabel: TypeAlias = Float[Label, "*batch"]
```

I think the * means "0 or more" so this is equivalent?

Reply:
Oh! I didn't know that. I think I misunderstood your suggestion, in that case. Then, let me see if I can make all the rules simpler!